dot/state/block_finalisation.go

```diff
@@ -72,6 +73,19 @@ func (bs *BlockState) GetFinalisedHash(round, setID uint64) (common.Hash, error)
 	return common.NewHash(h), nil
 }
 
+// retrieveFirstNonOriginBlockSlot returns the slot number of the first non origin block
+func (s *BlockState) retrieveFirstNonOriginBlockSlotFromDb() (uint64, error) {
+	slotVal, err := s.db.Get(firstSlotNumberKey)
+	if err != nil {
+		if errors.Is(err, database.ErrNotFound) {
+			return 0, nil
+		}
+		return 0, err
+	}
+	val := binary.LittleEndian.Uint64(slotVal)
+	return val, nil
+}
+
```

Comment:
@ramiroJCB could you move this method to `dot/state/block.go` below the method `SetFirstNonOriginSlotNumber` and change its name to `GetFirstNonOriginSlotNumber`?

Reply:
Done :)